src/c++/perf_analyzer/perf_analyzer_unit_tests.cc

```diff
@@ -0,0 +1,28 @@
+// Copyright 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

Comment:
Looks confusing. Why do we need this file? I have glanced over doctest. Doesn't the whole point of doctest is that you don't have to write the tests separately?
Why can't we include doctest.h and #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN in your new main.cc file?

Reply:
This goes back to the thread between you, Katherine, and me. If our CI was able to use a QA build of Perf Analyzer, then it would be feasible to recombine these, because then we could disable the compilation of unit tests for the Release build with the `DOCTEST_CONFIG_DISABLE` compile definition.

But since we don't have a separate build for QA and Release, if we don't want to release a version of Perf Analyzer that has compiled/accessible unit tests in it, we need separate executables: one for the release Perf Analyzer, and one for the unit tests.

Having this file is a way to create an executable for the unit tests, separate from the release Perf Analyzer executable.

If I include `doctest.h` and this `#define` in the new `main.cc`, then:
1) Perf Analyzer code wouldn't have a way to run, it would just run the unit tests
2) If I added a custom main function to call both the unit tests and perf analyzer, then we're back at the problem above where we would be releasing to customers a version of Perf Analyzer that contains compiled/accessible unit tests, and that seems unadvisable.

Reply:
Thanks for the explanation. Can you add few lines just before L27 explaining the utility of this file?

Something like the comment that you resolved in this file with little more explanation.

Reply:
Added comment 👍.